swiftui/ChartboostMediationDemo/Ads/Interstitial/InterstitialAdController.swift

```diff
@@ -44,47 +50,105 @@ class InterstitialAdController: NSObject, ObservableObject {
             return
         }
 
-        // Create an interstitial ad from the Chartboost Mediation ad provider.
-        guard let interstitialAd = chartboostMediation.interstitialAdProvider(with: self, andPlacementName: placementName) else {
-            // It could fail if the placement name is invalid.
-            print("[Error] failed to create interstitial advertisement for placement '\(placementName)'")
-            return
-        }
-        self.interstitialAd = interstitialAd
-
         // Notify the demo UI
         activityState = .running
 
-        // Associate any provided keywords with the advertisement before it is loaded.
-        interstitialAd.keywords = keywords
+        if fullscreenAPI {
+            // loadFullscreenAd expects keywords to be `[String : String]` instead of `HeliumKeywords?`.
+            let keywords = keywords?.dictionary ?? [:]
+            let request = ChartboostMediationAdLoadRequest(placement: placementName, keywords: keywords)
+            // Load the fullscreen ad, which will make a request to the network. Upon completion, a
+            // ChartboostMediationFullscreenAdLoadResult will be passed to the completion block.
+            chartboostMediation.loadFullscreenAd(with: request) { [weak self] result in
+                guard let self = self else { return }
+                if let ad = result.ad {
+                    ad.delegate = self
+                    self.fullscreenAd = result.ad
 
-        // Load the interstitial ad, which will make a request to the network. Upon completion, the
-        // delegate method `heliumInterstitialAd(withPlacementName:didLoadWithError:)` will be called.
-        interstitialAd.load()
+                    log(action: "load", placementName: placementName, error: result.error)
+                    // Notify the demo UI
+                    activityState = .idle
+                } else {
+                    fullscreenAd = nil
```

Comment:
At the beginning of this method there is a safeguard to prevent multiple taps on the load button from triggering multiple loads if a load is already ongoing.
See the line `guard interstitialAd == nil else...`
This doesn't work anymore for the fullscreen APIs. I suggest we create a flag (e.g. `private var isLoading = false`) and use it instead.

Reply:
I think I see what you're saying: when I was managing both the `interstitialAd` and `fullscreenAd` in the same controller, I didn't extend the safeguard used here to also cover `fullscreenAd`. So if I were keeping those two different ad types in the same class, it would make sense to use a `isLoading` that worked for both of them.

But since `fullscreenAd` is split off into its own file now, I think it's ok to just keep the original logic?

Reply:
Definitely. Just make sure the new file has some similar logic.